docs/framework/concepts/yaml_format.md

```diff
@@ -0,0 +1,283 @@
+
+# Features
+
+This YAML format exposes the framework API and provides helpers to write these apps descriptions more easily.
+
+## Shortdef
+
+To write more readable and compact apps descriptions, the format includes a `shortdef` keyword. The example below is identical to the previous one, except that it uses the keyword:
+
+```
+---
+root:
+  type: Controller.Mesh
+  slots:
+    window:
+      type: Window.Simple
+      slots:
+        content:
+          type: Pager.Simple
+          slots:
+            home-page:
+              shortdef: 'ContentGroup.ContentGroup(expand: true)'
+              slots:
+                arrangement:
+                  shortdef: 'Arrangement.List(expand: true)'
+                  slots:
+                    card: 'Card.List(expand: true)'
+                selection:
+                  type: Selection.All
+                  slots:
+                    filter: Filter.Articles
+                    order: Order.Sequence
+```
+
+In this example, the `type` and `properties` keywords are combined into single lines with the help of the `shortdef` keyword, e.g. _shortdef: 'Arrangement.List(expand: true)'_. When no slots are needed, the `shortdef` keyword can be omitted, using the short description directly on the slot name, e.g. _card: 'Card.List(expand: true)'_.
+
+## References
+
+Some modules need access to other modules properties. To do this, the format includes the `id` and `references` keywords, see the example below:
+
+```
+---
+root:
+  type: Controller.Mesh
+  slots:
+    window:
+      type: Window.Simple
+      slots:
+        content:
+          type: Pager.Simple
+          slots:
+            home-page:
+              shortdef: 'Layout.InfiniteScrolling(expand: true)'
+              references:
+                lazy-load: all-articles
+              slots:
+                content:
+                  shortdef: 'ContentGroup.ContentGroup(expand: true)'
+                  slots:
+                    arrangement:
+                      shortdef: 'Arrangement.List(expand: true)'
+                      slots:
+                        card: 'Card.List(expand: true)'
+                    selection:
+                      type: Selection.All
+                      id: all-articles
+                      slots:
+                        filter: Filter.Articles
+                        order: Order.Sequence
+```
+
+In this example a _Layout.InfiniteScrolling_ module is added, which needs access to the _Selection.All_ module. To do this, the selection module defines its identifier "all-articles" using the `id` keyword. Then, the scrolling module `references` the selection using the identifier. The "lazy-load" reference name is what the scrolling module uses to access this reference. The list of available references names depends on each type of module.
```

Comment:
Layout.InfiniteScrolling, Selection.All: link to the page in the API reference.
"all-articles", "lazy-load": use backticks instead of double quotes.